libs/gretty/src/main/groovy/org/akhikhl/gretty/GrettyPlugin.groovy

```diff
@@ -852,18 +852,17 @@ class GrettyPlugin implements Plugin<Project> {
 
     addExtensions(project)
     addConfigurations(project)
+    project.afterEvaluate {
+      afterProjectEvaluate(project)
+    }
 
+    //there can be multiple submodules applying this plugin and we want to register callback only once
     if(!project.rootProject.hasProperty('gretty_')) {
       Project rootProject = project.rootProject
       rootProject.ext.gretty_ = [:]
-      rootProject.ext.gretty_.evalProjectCount = rootProject.allprojects.sum 0, { it.state.executed ? 0 : 1 }
-      for(def p in rootProject.allprojects)
-        p.afterEvaluate { proj ->
-          afterProjectEvaluate(proj)
-          rootProject.ext.gretty_.evalProjectCount = rootProject.ext.gretty_.evalProjectCount - 1
-          if(rootProject.ext.gretty_.evalProjectCount == 0)
-            afterAllProjectsEvaluate(rootProject)
-        }
+      rootProject.getGradle().projectsEvaluated {
+        afterAllProjectsEvaluate(rootProject)
+      }
```

Comment:
I have no evidence that the "gretty_" property is used anywhere else. Also, I think that an empty map as sentinel value is... at least weird. Please see if you like the following approach better:

```groovy
Project rootProject = project.rootProject
Closure once = { afterAllProjectsEvaluate(rootProject) }.memoize()
rootProject.gradle.projectsEvaluated(once)
```

which would replace the entire block starting at `if(!project...)` and would get rid of the `gretty_` extension property.

Reply:
So we leave it as is. It's likely a bad idea...